python/setup.py

```diff
@@ -82,18 +91,81 @@
     "tune": ["tabulate"],
 }
 
+def is_native_windows_or_msys():  # Does NOT return true for WSL, which is considered "Linux"
+    return sys.platform == 'msys' or sys.platform == 'win32'
+
+def is_invalid_windows_platform():
+    return sys.platform == 'msys' or sys.platform == 'win32' and sys.version_string and 'GCC' in sys.version_string
+
+def makedirs(path, exist_ok):
```

Comment:
Please add a docstring comment specifying that this is just for python2 compatibility